contracts/Minter.sol

```diff
@@ -100,32 +104,62 @@ contract SPSMinter {
       token.mint(pools[i].receiver, amount);
 
       emit Mint(pools[i].receiver, amount);
+
+      unchecked { ++i; }
     }
   }
 
   /**
    * @notice Add new pool, can be called by admin
    * @param newReceiver Address of the receiver
    * @param newAmount Amount of tokens per block
+   * @param newReductionBlocks Number of blocks between emission reduction
+   * @param newReductionPercentage Percentage to reduce emission
    */
-  function addPool(address newReceiver, uint256 newAmount) external onlyAdmin {
+  function addPool(address newReceiver, uint256 newAmount, uint256 newReductionBlocks, uint256 newReductionPercentage) external onlyAdmin {
     require(pools.length < poolsCap, 'SPSMinter: Pools cap reached');
     require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
-    pools.push(Pool(newReceiver, newAmount));
-    emit PoolAdded(newReceiver, newAmount);
+    pools.push(Pool(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number));
+    emit PoolAdded(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number);
   }
 
   /**
    * @notice Update pool, can be called by admin
    * @param index Index in the array of the pool
    * @param newReceiver Address of the receiver
    * @param newAmount Amount of tokens per block
+   * @param newReductionBlocks Number of blocks between emission reduction
+   * @param newReductionPercentage Percentage to reduce emission
    */
-  function updatePool(uint256 index, address newReceiver, uint256 newAmount) external onlyAdmin {
+  function updatePool(uint256 index, address newReceiver, uint256 newAmount, uint256 newReductionBlocks, uint256 newReductionPercentage) external onlyAdmin {
     require(newAmount <= maxToPoolPerBlock, 'SPSMinter: Maximum amount per block reached');
     mint();
-    pools[index] = Pool(newReceiver, newAmount);
-    emit PoolUpdated(index, newReceiver, newAmount);
+    pools[index] = Pool(newReceiver, newAmount, newReductionBlocks, newReductionPercentage, block.number);
```

Comment:
There should be a require statement that `newReductionPercentage` is less than or equal to 100, unless a *growing* share of tokens is something we want to explicitly support? Even then, there should be some check bounding check.

Reply:
Resolved: https://github.com/steem-monsters/sps-minter/commit/0d13256d6be6ffc532195bc1489969a9517f3536